lib/bap_disasm/bap_disasm_rec.ml

```diff
@@ -299,6 +299,46 @@ let create_indexes (dests : dests Addr.Table.t) =
 
 let filter_valid s = {s with inits = Set.inter s.inits s.valid}
 
+let join_destinations dests =
```

Comment:
I think that we can stick to a more simple spraying destination in case of the nondeterminism and adopt the following scheme:
```
if (unk) jmp d1;
...
if (unk) jmp dm;
jmp d;
```

where `d` is the original BIL destination and `d1,dm` is the set of destinations provided by the brancher (alternatively you can just pick an arbitrary `d` from the whole set, it doesn't really matter).

This code will produce graphs that are simple (ideally, it should map to a single block with multiple `when` guarded jumps), while your version requires some sophistication during graph generation and with naive approach will spill a graph that is quadratic with respect to the number of destinations (which in case if we will retarget VSA here, could be a very big number).